content/blog/vscode-devcontainers/index.md

```diff
@@ -0,0 +1,268 @@
+---
+title: "Build a Pulumi VS Code Devcontainer Environment"
+date: 2020-07-14
+meta_desc: "Build a VS Code devcontainer for Pulumi infrastructure as code projects."
+meta_image: devcontainer.png
+authors:
+    - sophia-parafina
+tags:
+    - "VS Code"
+    - "devcontainers"
+    - Docker
+---
+
+One of the major advantages of using containers for development is reducing the need to install software and all of the associated dependencies. Developers can start writing code without configuring a development environment that emulated production. The Visual Studio Code Remote - Containers extension lets you develop inside a container. If you want to use Pulumi’s infrastructure as code engine without having to install the Pulumi CLI, this blog post is for you!
+
+<!--more-->
+
+## Pre-requisites
+
+To set up your development environment you’ll need Visual Studio Code, [Docker Desktop](https://www.docker.com/products/docker-desktop), and the [Remote - Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers).
+
+1. Check to see if Docker is running:
+
+    ```bash
+    $ docker --version
+    Docker version 19.03.8, build afacb8b
+    ```
+
+1. Check to see if the **Remote - Containers** is installed:
+
+    ![Remote status](remote-status-bar.png)
+
+    Click on the green icon in the Remote Status Bar to display the commands:
+
+    ![Remote container commands](remote-containers-commands.png)
+
+## Building the Pulumi container
+
+Now we can start!
+
+1. Create a directory for your project.
+
+    ```bash
+    $ mkdir my_devcontainer
+    $ cd my_devcontainer
+    ```
+
+    Open the directory in VS Code.
+
+    ![Open directory](open_devcontainer_dir.png)
+
+1. Next, we’ll create the devcontainer configuration files. Click on the Remote Status button and select **Remote Container - Add Development Container Configuration Files ...**.
+
+    ![A configuration files](add_devcontainer_config_files.png)
+
+1. Select **Debian 10 & Git**. We’ll replace the Dockerfile with the Pulumi Dockerfile.
+
+    ![Select config file](select_config_file.png)
+
+    VS Code will display a popup asking to build the container. We’ll replace the Dockerfile with the Pulumi Dockerfile, dismiss the popup.
+
```

Comment:
I might reverse the order of the last sentence "Dismiss the popup, we will replace the Dockerfile with the Pulumi Dockerfile."  I have a bad habit of clicking before I get to the end of sentences though.